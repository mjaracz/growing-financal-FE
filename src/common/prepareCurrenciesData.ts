import { CurrenciesResponse } from 'molecules/currencySelect';
import { CurrencyRate } from 'molecules/currencySelect';

export function prepareCurrenciesData(currenciesResponse: CurrenciesResponse): CurrencyRate[] {
  const currenciesKeys: string[] = Object.keys(currenciesResponse.rates);
  const currenciesValues: number[] = Object.values(currenciesResponse.rates);
  const preparedCurrencies = [];
  for (let i = 0; i < currenciesKeys.length; i++) {
    const currencyRate = {}
    currencyRate[currenciesKeys[i]] = currenciesValues[i];
    preparedCurrencies.push(currencyRate)
  }
  return preparedCurrencies;
}