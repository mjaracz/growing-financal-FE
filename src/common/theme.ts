import { createMuiTheme } from '@material-ui/core';

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: 'hsl(193, 95%, 68%)',
      dark: 'hsl(220, 13%, 18%)'
    },
    secondary: {
      main: 'hsl(281, 100%, 80%)'
    },
    background: {
      default: 'hsl(220, 13%, 18%)'
    },
    text: {
      secondary: 'hsl(0, 0%, 100%)',
      hint: 'rgb(215, 146, 247)'
    }  
  }
})