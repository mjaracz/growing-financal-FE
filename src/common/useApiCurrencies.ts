export const useApiCurrencies = () => {
  const baseUrl = 'http://api.exchangeratesapi.io/v1/latest';
  const apiKey = '481b03536817ae9a3877e3026e237f51';

  const getLatestRate = async () => 
    fetch(baseUrl + `?access_key=${apiKey}`)
      .then(res => res.json())
      .then((data) => {
        if (data.statusCode >= 400) {
          throw new Error('User code error 4** status')
        }
        return data;
      })

  return {
    getLatestRate,
    baseUrl,
    apiKey,
  }
}
