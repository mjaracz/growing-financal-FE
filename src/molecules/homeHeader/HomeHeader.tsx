import React from 'react';
import { useStyle } from './useStyle';

export const HomeHeader = () => {
  const styles = useStyle();
  return (
    <header className={styles.homeHeader}>
      
    </header>
  )
}