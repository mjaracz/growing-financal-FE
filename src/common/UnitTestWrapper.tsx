import React, { FC } from 'react';
import { ThemeProvider } from '@material-ui/core';
import { theme } from './theme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from 'redux/store';

export const UnitTestWrapper: FC = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          {children}
        </Provider>
      </ThemeProvider>
    </BrowserRouter>
  )
}