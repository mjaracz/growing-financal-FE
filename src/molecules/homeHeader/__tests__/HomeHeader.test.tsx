import React from 'react';
import { render } from '@testing-library/react';
import { HomeHeader } from '../HomeHeader';
import { UnitTestWrapper } from 'common/UnitTestWrapper';

describe('<HomeHeader />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <UnitTestWrapper>
        <HomeHeader />
      </UnitTestWrapper>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
