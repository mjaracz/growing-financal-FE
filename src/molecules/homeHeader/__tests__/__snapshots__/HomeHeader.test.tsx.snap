// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeHeader /> should match the snapshot 1`] = `
<body>
  <div>
    <header
      class="makeStyles-homeHeader-1"
    >
      <img
        alt="logo"
        class="makeStyles-headerLogo-2"
        src="reactLogo.svg"
      />
    </header>
  </div>
</body>
`;
